File: labs OOP/2.2/C++/Numbers.cpp
#include "Numbers.h"

Numbers::Numbers(const std::string& s) : value(s) {}

int Numbers::length() const {
    return static_cast<int>(value.length());
}

int Numbers::position(char c) const {
    size_t pos = value.find(c);
    return (pos != std::string::npos) ? static_cast<int>(pos) : -1;
}

std::string Numbers::getValue() const {
    return value;
}

void Numbers::shift() {
    if (!value.empty()) {
        char last = value.back();
        value = std::string(1, last) + value.substr(0, value.length() - 1);
    }
}

File: labs OOP/2.2/C#/Numbers.cs
using System;

namespace StringPolymorphism
{
    public class Numbers : Strings
    {
        private string value;

        public Numbers(string s)
        {
            value = s;
        }

        public override int Length()
        {
            return value.Length;
        }

        public override int Offset(char c)
        {
            int pos = value.IndexOf(c);
            return pos;
        }

        public override string GetValue()
        {
            return value;
        }

        public override void Shift()
        {
            if (!string.IsNullOrEmpty(value))
            {
                char last = value[^1];
                value = last + value.Substring(0, value.Length - 1);
            }
        }
    }
}